google-cloud-spanner-hibernate-tools/src/main/java/com/google/cloud/spanner/hibernate/PooledBitReversedSequenceStyleGenerator.java

```diff
@@ -414,6 +414,7 @@ private Iterator<Long> fetchIdentifiers(SharedSessionContractImplementor session
         connection.commit();
         statement.execute(String.format("set %sretry_aborts_internally=false", extensionPrefix));
         List<Long> identifiers = new ArrayList<>(this.fetchSize);
+        session.getJdbcServices().getSqlStatementLogger().logStatement(this.select);
```

Comment:
Are we sure we want to add this log statement?

Reply:
> do you see any issues with logging this ?

No, nothing specific, other than making sure that we prevent unnecessary log spamming. Your reasoning that this is something that you want to see when you enable show_sql sounds good.